src/Sarif.Converters/AndroidStudioConverter.cs

```diff
@@ -150,51 +152,73 @@ public Result ConvertProblemToSarifResult(AndroidStudioProblem problem)
+                // 
+                //if (foundUriBaseId)
+                //{
+                //    location.PhysicalLocation.FileLocation.UriBaseId = PROJECT_DIR;
+                //}
             }
 
             result.Locations = new List<Location> { location };
 
             return result;
         }
 
-        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem)
+        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem, out int index)
         {
-            string parentLogicalLocationKey = null;
+            index = -1;
+            string fullyQualifiedName = null;
+            string delimiter = string.Empty;
+
+            if (!string.IsNullOrEmpty(problem.Module))
+            {
+                index = AddLogicalLocation(index, ref fullyQualifiedName, problem.Module, LogicalLocationKind.Module, delimiter);
+                delimiter = @"\";
+            }
 
-            parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.Module, LogicalLocationKind.Module);
-            parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.Package, LogicalLocationKind.Package);
+            if (!string.IsNullOrEmpty(problem.Package))
+            {
+                index = AddLogicalLocation(index, ref fullyQualifiedName, problem.Package, LogicalLocationKind.Package, delimiter);
+                delimiter = @"\";
+            }
 
             if (problem.EntryPointName != null)
             {
                 if ("class".Equals(problem.EntryPointType, StringComparison.OrdinalIgnoreCase))
                 {
-                    parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.EntryPointName, LogicalLocationKind.Type);
+                    index = AddLogicalLocation(index, ref fullyQualifiedName, problem.EntryPointName, LogicalLocationKind.Type, delimiter);
+
                 }
                 else if ("method".Equals(problem.EntryPointType, StringComparison.OrdinalIgnoreCase))
                 {
-                    parentLogicalLocationKey = AddLogicalLocation(parentLogicalLocationKey, problem.EntryPointName, LogicalLocationKind.Member);
+                    index = AddLogicalLocation(index, ref fullyQualifiedName, problem.EntryPointName, LogicalLocationKind.Member, delimiter);
                 }
+                // TODO: 'file' is another entry point type. should we be doing something here?
             }
 
-            return parentLogicalLocationKey;
+            return fullyQualifiedName;
         }
 
-        private string AddLogicalLocation(string parentKey, string value, string kind, string delimiter = @"\")
+        private int AddLogicalLocation(int parentIndex, ref string fullyQualifiedName, string value, string kind, string delimiter = ".")
         {
-            if (!String.IsNullOrEmpty(value))
+            fullyQualifiedName = fullyQualifiedName + delimiter + value;
```

Comment:
>fullyQualifiedName [](start = 33, length = 18)

How does this work on the first call, where `fullyQualifiedName` comes in as `null`? I could see this working if it was initialized to `string.Empty` on Line 175.

Reply:
.NET string.concat happily accepts null and treats these as, effectively, string.empty instances.

---
In reply to: [243866667](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243866667) [](ancestors = 243866667)